docs/copied-from-beats/docs/security/users.asciidoc

```diff
@@ -197,156 +344,245 @@ information.
 +
 [options="header"]
 |====
 |Role | Purpose
 
 |`monitoring_user`
-|Use *Stack Monitoring* in {kib} to monitor {beatname_uc}
+|Grants access to monitoring indices for {beatname_uc}
 |====
---
-endif::serverless[]
 
-[[privileges-to-publish-events]]
-==== Grant privileges and roles needed for publishing
+////
+***********************************  ***********************************
+***********************************  ***********************************
+////
 
-Users who publish events to {es} need to create and write to {beatname_uc}
-indices. To minimize the privileges required by the writer role, use the
-<<privileges-to-setup-beats,setup role>> to pre-load dependencies. This section
-assumes that you've pre-loaded dependencies. 
+[[kibana-user-privileges]]
+=== Grant privileges and roles needed to read APM data in {kib}
 
-ifndef::no_ilm[]
-When using ILM, turn off the ILM setup check in the {beatname_uc} config file before
-running {beatname_uc} to publish events:
+++++
+<titleabbrev>Create a _reader_ user</titleabbrev>
+++++
 
-[source,yaml]
-----
-setup.ilm.check_exists: false
-----
-endif::no_ilm[]
+{kib} users typically need to view apps, dashboards, and visualizations that contain
+APM data. These users might also need to create and edit dashboards, visualizations, and machine learning jobs.
 
-To grant the required privileges:
+To grant users the required privileges:
 
-. Create a *writer role*, called something like +{beat_default_index_prefix}_writer+,
-that has the following privileges:
+. Assign the following built-in roles to users who need to read APM data:
++
+[options="header"]
+|====
+|Role | Purpose
+
+|`kibana_user`
+|Use the APM UI
+
+|`apm_user`
+|Use the APM UI
+|====
 +
-NOTE: The `monitor` cluster privilege and the `create_doc` privilege on
-+{beat_default_index_prefix}-*+ indices are required in every configuration.
+. If you're using any of the APM, Dashboard, Visualize, Discover, etc., apps,
+you'll need assign space privileges:
 +
 [options="header"]
 |====
 |Type | Privilege | Purpose
 
-ifndef::apm-server[]
-|Cluster
-|`monitor`
-|Retrieve cluster details (e.g. version) 
-endif::apm-server[]
+| Spaces
+| `Read` or `All` on {beat_kib_app}
+| Allow the use of the {beat_kib_app}
 
-ifndef::no_ilm[]
-|Cluster
-|`read_ilm`
-| Read the ILM policy when connecting to clusters that support ILM.
-Not needed when `setup.ilm.check_exists` is `false`.
-endif::no_ilm[]
+| Spaces
+| `Read` or `All` on Dashboards, Visualize, and Discover
+| Allow the user to view, edit, and create dashboards, as well as browse data.
+|====
++
+. Finally, assign the following role if a user needs to enable and edit machine learning features:
++
+[options="header"]
+|====
+|Role | Purpose
 
-ifeval::["{beatname_lc}"=="filebeat"]
-|Cluster
-|`cluster:admin/ingest/pipeline/get`
-|Check for ingest pipelines used by modules. Needed when using modules.
-endif::[]
+|`machine_learning_admin`
+|NEED DESCRIPTION
```

Comment:
for inspiration have a look at [built-in roles](https://www.elastic.co/guide/en/elasticsearch/reference/7.7/built-in-roles.html) description for `machine_learning_admin`. It needs to be set when users should be able to create ML jobs on monitored services.